python/pyarrow/array.pxi

```diff
@@ -538,6 +538,51 @@ def repeat(value, size, MemoryPool memory_pool=None):
     return pyarrow_wrap_array(c_array)
 
 
+def mask(indices, length, MemoryPool memory_pool=None):
+    """
+    Create a boolean Array instance where specific indices are marked as True.
+
+    Parameters
+    ----------
+    indices : list[int]
```

Comment:
If I remember correctly compute functions are expected to work _only_ on arrow data (arrays), which would make less convenient this function as a factory as its purpose is mostly to create a mask from literals.

I think it would make sense to make it accept arrays too btw

Reply:
At the C++ level you're right, at the Python level we allow implicit conversion if it's non-ambiguous:
```python
>>> pc.list_flatten([[1, 2, 3], None, [4, None, 5]])
<pyarrow.lib.Int64Array object at 0x7f37d6454340>
[
  1,
  2,
  3,
  4,
  null,
  5
]
```

@felipecrv What do you think about this? Should it be a compute kernel or rather a general facility?

Reply:
If the utility is templated on integer type, it could be used to implement kernels, but implementing the kernel involves a lot more code due to dispatching logic. So I would be in favor of having the utility as it is and leaving the kernel for later.

Reply:
Added an overloaded version that accepts Array